circle.yml

```diff
@@ -0,0 +1,21 @@
+machine:
+  xcode:
+    version: "8.0"
+
+dependencies:
+  pre:
+    - brew update 1> /dev/null 2> /dev/null
+    - brew outdated carthage || brew upgrade carthage
+
+checkout:
+  post:
+    - git submodule update --init --recursive
+test:
+  override:
+    - ./scripts/macOS
```

Comment:
@ZevEisenberg It's slightly too complex to do in shell alone and i don't want to mess with a ruby environment right now. I'm going to add "clean up build scripts" as an issue so it doesn't get lost and add the `.sh` extensions.